### DISCUSSION TITLE
How Much to Build a Bridge? (3 levels)

### DISCUSSION THREAD

**Discussion by Shadow8472** [smcameron/space-nerds-in-space#322] [General]
Cross Post from Reddit:
[https://www.reddit.com/r/SNIS/comments/wjsr3q/how_much_to_build_a_bridge_set/](https://www.reddit.com/r/SNIS/comments/wjsr3q/how_much_to_build_a_bridge_set/)

I'm dreaming of doing something a little more than a LAN party. I do believe the GPL2 license would allow me to put on a show?

Level 1: Fall Festival Fundraiser for my church's school
- Construction Budget: Scavenging equipment or begging parents for money. I'd consider myself lucky if I get away for <$1,000.
- Quality: A guided imagination can see the bridge for the folding tables/chairs, old, mismatched monitors, and cheap-o computers.
- Players: Minimum 4 (3 computers, 2 Pi's, and maybe a tablet)
- Special Effects: N/A

Level 2: Portable Set for Rent/Fair/Convention
- Construction Budget: No clue. I'll guess $5,000-$10,000.
- Quality: The U.S.S. Obscurity offers full Immersion on a budget. Navigation/weapons are on joysticks, other stations have custom control panels.
- Players: Up to a full crew
- Special Effects: Surround sound, extra lights, uniforms for rent/purchase, captain's tablet for secret missions, and custom officer training videos

Level 3: I've got a studio and I'm not afraid to use it! (I'm dreaming much larger than life here)
- Construction Budget: Depends on prior and continued success.
- Quality: Captain on Deck! The world of SNIS is brought to life like never before. Either elephants are small or you are aboard the U.S.S. Michal Collins. 
- Special Effects: vibrating deck plates, motion capture aliens to hail, exploding control panels, repair minigames, and a mechanism to tilt the whole bridge. Don't forget hidden cameras to make a cheesy space comedy/drama afterwords.

Level 3 is just a dream and likely will forever remain such. However, I am seriously considering 1 and intend to decide if I want to pursue 2 if it goes well.

**Comment by smcameron** [comment]
Well, how much it costs depends on how good you are at scavenging, and getting help from other people, I guess.

The cheapest way is to use OPC (Other People's Computers).

Assuming you don't have a bunch of Linux nerd friends, the cheapest way to do that is to get a handful of USB thumb drives, install linux and SNIS on each of these thumb drives, then get your friends to boot up their windows computers off the thumb drives.  You'll also need some CAT5 cable and a network switch (you don't want to try run everything over wireless, one or two stations on wireless might be ok, but not all of them.)  Assuming you and your friends already have enough computers, this shouldn't be more than, I don't know... $150-$200 should cover it easily, I'd think.  Hmm, maybe the biggest expense is you want a big screen, which means a dark room and a projector, or a really big TV.

It is also possible to run snis_client on WSL (windows system for linux), however, last I tried it, you could not get hardware graphics acceleration to work, (WSL2 may change this, but I haven't tried it) so the graphics heavy screens weren't fast enough to be acceptable.  Might be ok for SCIENCE, COMMS, ENGINEERING / DAMAGE CONTROL though.  Similarly Raspberry Pi 4 are kind of slow, so they're only good for those stations as well.

IF you want to get fancy (your level 2, I guess), and build dedicated consoles for some or all stations with cool toggle switches, lights, and buttons, knobs, and whatnot, I have sort of started down that path, at least designing, using Arduinos to monitor switch/button/potentiometer states and turn LEDs on/off, communicating with snis_client via USB serial, however, none of these consoles have actually been built, so there is also some software work that would need to be done as well (partially there, but not well tested.)  More details about that here: https://spacenerdsinspace.com/snis-consoles/notes.html   This all kind of got derailed by COVID.  The parts for all this is not too bad (last I checked, pre-COVID, might be worse now), speakers, amps, keyboards for Raspberry Pi sound can be had cheaply from aliexpress, capacitors, resistors, headers, switches, potentiometers, etc. from futurlec.com (slow, but cheap), knobs from taydaelectronics.com  The Raspberry Pis and monitors will be the most expensive parts.  As for building the physical consoles with the buttons, switches, etc. there are no designs for the physical enclosures etc. so all that work remains to be done.  This video might be of interest: https://www.youtube.com/watch?v=51ljGNx3px4

For level 3, I don't know how to price that out at all.  I guess it comes down to plywood, 2x4s, paint, foam, and a laser cutter mostly, I'd guess, and depends on just how crazy you want to go.  There are some star wars nerds doing interesting things, one example: https://www.youtube.com/watch?v=XAVey6wy-O4

**Reply by Shadow8472** [reply]
I'm pretty handy with multiple branches of Linux, but I haven't done a lot of manual compilation. Manjaro/AUR was my shortcut of choice in that respect. I'm planning on calling a computer recycling place to see if they might have some second hand equipment in working order I can pull from the scrap heap. Hopefully some local company is replacing all their workstations and I can score a matching set. I could see myself trying to arrange it as a donation to the school for a computer lab, but that would set me back in terms of building a rental. I'd need to go with a TV if I'm looking to do a booth where I cannot control the lighting.

Level 2 is --in all likelihood-- as far as I'll ever go. You're definitely on the bridge of a used space ship, so perfection isn't mandatory.

Level 3 is what could happen if Level 2 is wildly successful. It's meant to grow as budget increases.

**Reply by smcameron** [reply]
Compiling  SNIS on a debian based distro (e.g. ubuntu is the most common) is pretty easy.  It's just a bunch of "apt-get install blah blah blah" to get the prerequisites, followed by "make", then "make update-assets".  Most problems people run into are caused by them accidentally missing a dependency.

If you're trying to run this as a business... well, good luck.  I suppose it might work, but it seems pretty risky to me.  I think other people have tried similar things with Artemis, but I don't know if any have been successful.  You might look here for inspiration: https://artemis.forumchitchat.com/post/2285inspired-control-panels-8368041  and also here https://artemis.forumchitchat.com/post/armada-v-photos-12413504  

Have you played SNIS yet?  If not, you probably want to try it out before spending a lot of money trying to start a business based on it.  If you don't currently have the equipment to pull it off, and if you happen to have a hackerspace in your area, then I'd suggest hanging out there for awhile -- most of them have an open house once a week or so -- and trying to get some people there interested in trying it out.  The kind of people that will already have linux on their laptops and be into star trek will be there, and there will probably already be a projector there, so it's a good place to try to find a crew for a few test drives.  If there's no nearby hackerspace, then it's harder. 

You might look at these guys: http://bridgesim.arkgeeks.com/   I don't know if that's a business or nerd club, or what.

**Reply by Shadow8472** [reply]
I'm trying to brainstorm ideas on how I can monetize my technology blog while remaining true to my principles as a privacy advocate.

I played Artimis a few times around 10 years ago and tried my hand at operating a SNIS ship solo with the intention of doubling up stations until I can find enough people I know who are willing to crew them. I managed to get it down to 3, but I'm a little iffy on one or two of the combo stations.

**Reply by smcameron** [reply]
You can certainly play it solo, flipping between stations as needed, but of course much of the fun of this type of game is not even in the game, but in the people playing it.

**Reply by kwadroke** [reply]
> You might look at these guys: http://bridgesim.arkgeeks.com/ I don't know if that's a business or nerd club, or what.

That's my website :) Not a business, sort of a nerd club. Mostly just getting the word out about BridgeSims at local Conventions and finding others to play with.

**Reply by smcameron** [reply]
Ha, yeah, I knew it was your website, just didn't know exactly what kind of an organization it was supposed to be.  Edit: And in case it isn't obvious, "nerd club" is not meant to be derogatory in any way.  From 2010-2014, every Friday night, me and my best friend would go to the hackerspace in Houston, TX/RX Labs, to hang out, and we affectionately referred to it as "nerd club."

**Reply by smcameron** [reply]
Also, tonight, I went to the hackerspace where I am now, in Richmond, VA, called HackRVA.  A couple of years ago, I was working with a guy there and we were going to build some boards, a la what's described here: https://spacenerdsinspace.com/snis-consoles/notes.html  and then COVID came along and squashed everthing ... well I saw this guy tonight, and brought it up again, and am trying to start this up again.

**Reply by kwadroke** [reply]
> Ha, yeah, I knew it was your website, just didn't know exactly what kind of an organization it was supposed to be. Edit: And in case it isn't obvious, "nerd club" is not meant to be derogatory in any way. From 2010-2014, every Friday night, me and my best friend would go to the hackerspace in Houston, TX/RX Labs, to hang out, and we affectionately referred to it as "nerd club."

I didn't view it as derogatory. It does have the word "Geek" in the Name & Domain after all.

**Comment by smcameron** [comment]
> I do believe the GPL2 license would allow me to put on a show?

It is certainly allowed. You can put SNIS in a show, film, play, exhibition, arcade, whatever, no problem.
